src/states/node.rs

```diff
@@ -2458,3 +2476,336 @@ fn create_first_section_info(public_id: PublicId) -> Result<SectionInfo, Routing
         err
     })
 }
+
+#[cfg(all(test, feature = "mock_parsec"))]
+mod tests {
+    use super::*;
+    use crate::cache::NullCache;
+    use crate::mock_crust::crust::Config;
+    use crate::mock_crust::{self, Network};
+    use crate::outbox::{EventBox, EventBuf};
+    use crate::state_machine::{State, StateMachine};
+    use crate::xor_name::XOR_NAME_LEN;
+    use utils::LogIdent;
+
+    const ACCUMULATE_VOTE_COUNT: usize = 3;
+    const NOT_ACCUMULATE_ALONE_VOTE_COUNT: usize = 2;
+
+    struct NoteUnderTest {
+        pub machine: StateMachine,
+        pub full_id: FullId,
```

Comment:
I'd consider adding a `full_id` method to `StateMachine` (test-only) so you don't have to duplicate it here (and possibly risking them getting out of sync).

Reply:
The risk is super low, since it is test code that would likely fail if it happened. Not sure it is worth adding noise to the StateMachine yet.

Reply:
I think it's a good practice to reduce the duplication of information. Probably not a big issue here yet though.